includes/class-wcsr-resource.php

```diff
@@ -187,24 +188,84 @@ public function get_days_active( $from_timestamp, $to_timestamp = null ) {
 		$activation_times   = self::get_timestamps_between( $this->get_activation_timestamps(), $from_timestamp, $to_timestamp );
 		$deactivation_times = self::get_timestamps_between( $this->get_deactivation_timestamps(), $from_timestamp, $to_timestamp );
 
-		// Now find the number of days between the timestamps where the resource was active (the resource must be inactive for more than 24 hours to be considered inactive for a given day)
-		$days_active = floor( ( $to_timestamp - $from_timestamp ) / DAY_IN_SECONDS );
+		// if the first activation date is after the first deactivation date, make sure we prepend the start timestamp to act as the first "activated" date for the resource
+		if ( ! isset( $activation_times[0] ) || ( isset( $deactivation_times[0] ) && $activation_times[0] > $deactivation_times[0] ) ) {
+			$start_timestamp = ( $this->get_date_created()->getTimestamp() > $from_timestamp ) ? $this->get_date_created()->getTimestamp() : $from_timestamp;
 
-		// First remove any gap between the $from_timestamp and the resources creation date
-		if ( $this->get_date_created()->getTimestamp() > $from_timestamp ) {
-			$days_active -= floor( ( $this->get_date_created()->getTimestamp() - $from_timestamp ) / DAY_IN_SECONDS );
+			// before setting the start timestamp as the created time or the $from_timestamp make sure the deactivation date doesn't come before it
+			if ( isset( $deactivation_times[0] ) && $start_timestamp > $deactivation_times[0] ) {
+				throw new Exception( 'The resource first deactivation date in the period comes before the resource start time or before the beginning of the period. This is invalid.' );
+			}
+
+			array_unshift( $activation_times, $start_timestamp );
 		}
 
 		foreach ( $activation_times as $i => $activation_time ) {
-			if ( isset( $deactivation_times[ $i ] ) ) {
-				$days_active -= floor( ( $activation_time - $deactivation_times[ $i ] ) / DAY_IN_SECONDS );
+			// If there is corresponding deactivation timestamp, the resource has deactivated before the end of the period so that's the time we want, otherwise, use the end of the period as the resource was still active at end of the period
+			$deactivation_time = isset( $deactivation_times[ $i ] ) ? $deactivation_times[ $i ] : $to_timestamp;
+
+			// skip over any days that are activated/deactivated on the same day and have already been accounted for
```

Comment:
I wonder if we should remove the use of `same day` everywhere to avoid any confusion. Instead, it could use something like "same 24 hour block".

As covered nicely [here](https://github.com/Prospress/woocommerce-subscriptions-resource/pull/12/files#diff-707db6cb1b4d41cc9853cf9428a4228dR239), day is ambiguous - it can be a calendar day, or a 24 hour period. Explicitly using "_24 hour block_" isn't perfect, because it doesn't capture the fact that each 24 hours period has a set base it is calculated from, so it is still ambiguous, but at least it doesn't have a clear, more commonly used definition we're not honouring like "day" does with calendar day. Thoughts?

Reply:
I agree, saying 24hour block instead of same day definitely helps with any confusions i have.. I'll update it across the board to use the same terminology

Reply:
This has been fixed up with aab43e6 (not sure why it's still referencing line 207 and not updating after the rebase)